internal/services/appconfiguration/app_configuration_key_resource.go

```diff
@@ -0,0 +1,274 @@
+				return fmt.Errorf("while parsing resource ID: %+v", err)
+			}
+
+			client, err := metadata.Client.AppConfiguration.DataPlaneClient(ctx, resourceID.ConfigurationStoreId)
+			if err != nil {
+				return err
+			}
+
+			res, err := client.GetKeyValues(ctx, resourceID.Key, resourceID.Label, "", "", []string{})
+			if err != nil {
+				if !utils.ResponseWasNotFound(res.Response().Response) {
+					return metadata.MarkAsGone(resourceID)
+				}
+				return fmt.Errorf("while checking for key's %q existence: %+v", resourceID.Key, err)
+			}
+
+			if len(res.Values()) > 1 {
+				return fmt.Errorf("unexpected API response. More than one value returned for Key/Label pair %s/%s", resourceID.Key, resourceID.Label)
+			}
+
+			kv := res.Values()[0]
+			var locked bool
+			if kv.Locked != nil {
+				locked = *kv.Locked
+			}
+
+			model := KeyResourceModel{
+				ConfigurationStoreId: resourceID.ConfigurationStoreId,
+				Key:                  utils.NormalizeNilableString(kv.Key),
+				ContentType:          utils.NormalizeNilableString(kv.ContentType),
+				Label:                utils.NormalizeNilableString(kv.Label),
+				Value:                utils.NormalizeNilableString(kv.Value),
+				Locked:               locked,
+				Tags:                 tags.Flatten(kv.Tags),
+			}
+
+			return metadata.Encode(&model)
+		},
+		Timeout: 5 * time.Minute,
+	}
+}
+
+func (k KeyResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+
+			resourceID, err := parse.AppConfigurationKeyID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("while parsing resource ID: %+v", err)
+			}
+
+			client, err := metadata.Client.AppConfiguration.DataPlaneClient(ctx, resourceID.ConfigurationStoreId)
+			if err != nil {
+				return err
+			}
+
+			var model KeyResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			if metadata.ResourceData.HasChange("value") || metadata.ResourceData.HasChange("content_type") || metadata.ResourceData.HasChange("tags") {
+
+				entity := appconfiguration.KeyValue{
+					Key:         utils.String(model.Key),
+					Label:       utils.String(model.Label),
+					ContentType: utils.String(model.ContentType),
+					Value:       utils.String(model.Value),
+					Tags:        tags.Expand(model.Tags),
+				}
+
+				_, err = client.PutKeyValue(ctx, model.Key, model.Label, &entity, "", "")
+				if err != nil {
+					return fmt.Errorf("while updating key/label pair %s/%s: %+v", model.Key, model.Label, err)
+				}
+			}
+
+			if metadata.ResourceData.HasChange("locked") {
+				if model.Locked {
+					_, err = client.PutLock(ctx, model.Key, model.Label, "", "")
+					if err != nil {
+						return fmt.Errorf("while locking key/label pair %s/%s: %+v", model.Key, model.Label, err)
+					}
+				} else {
+					_, err = client.DeleteLock(ctx, model.Key, model.Label, "", "")
+					if err != nil {
```

Comment:
minor but we can make this:

```suggestion
					if _, err = client.DeleteLock(ctx, model.Key, model.Label, "", ""); err != nil {
```